horreum-web/src/domain/runs/DatasetComparison.tsx

```diff
@@ -213,28 +214,54 @@ type ViewComparisonProps = {
 
 function ViewComparison({headers, view, datasets, alerting}: ViewComparisonProps) {
     const [loading, setLoading] = useState(false)
-    const [rows, setRows] = useState<IRow[]>()
+    const [rows, setRows] = useState<IRow[]>([])
     const token = useSelector(tokenSelector)
+
     useEffect(() => {
         setLoading(true)
         Promise.all(datasets.map(ds => datasetApi.getSummary(ds.id, view.id)))
             .then(
                 summaries => {
-                    setRows(
-                        view.components.map(vc => ({
-                            cells: [
-                                vc.headerName,
-                                ...summaries.map(summary => {
-                                    const render = renderValue(
-                                        vc.render,
-                                        vc.labels.length == 1 ? vc.labels[0] : undefined,
-                                        token
-                                    )
-                                    return render(summary.view?.[vc.id], summary)
-                                }),
-                            ],
-                        }))
-                    )
+                    const rowsData: any[][] = view.components.map(vc => {
+                        const rowCells = [vc.headerName];
+                        summaries.forEach(summary => {
+                            const render = renderValue(vc.render, vc.labels.length === 1 ? vc.labels[0] : undefined, token);
+                            rowCells.push(render(summary.view?.[vc.id], summary));
+                        });
+                        return rowCells;
+                    });
+                    const allRows: IRow[] = []
+
+                    rowsData.forEach((rd, index) => {
+                        const isNum = typeof rd[rd.length -1] === 'number'
+                        allRows.push({
+                            isOpen: isNum ? false: undefined,
+                            cells:  rd.map( (r, index) => ({
+                                    title: r
```

Comment:
I think `r` could be a json object which probably cannot be directly injected into the cell.